src/mappings/storage/utils.ts

```diff
@@ -206,3 +214,134 @@ export async function deleteDataObjectsByIds(
   subtitlesRepository.remove(...currentSubtitles.flat())
   await deleteDataObjects(overlay, block, indexInBlock, extrinsicHash, objects)
 }
+
+export async function processSetNodeOperationalStatusMessage(
+  overlay: EntityManagerOverlay,
+  block: Block,
+  indexInBlock: number,
+  extrinsicHash: string | undefined,
+  workingGroup: 'storageWorkingGroup' | 'distributionWorkingGroup',
+  actor: Worker | undefined, // worker who sent the message, undefined if it's a lead
+  meta: ISetNodeOperationalStatus
+): Promise<void> {
+  let workerId: bigint
+  const bucketId = meta.bucketId || ''
+
+  // Get the bucket's worker id
+  if (actor) {
+    workerId = actor.runtimeId
+  } else {
+    const maybeWorker = await overlay
+      .getRepository(Worker)
+      .getById(`${workingGroup}-${meta.workerId}`)
+
+    if (!maybeWorker) {
+      return invalidMetadata(
+        SetNodeOperationalStatus,
+        `The worker ${meta.workerId} does not exist in the ${workingGroup} working group`
+      )
+    }
+    workerId = maybeWorker.runtimeId
+  }
+
+  // Update the operational status of Storage node
+  if (workingGroup === 'storageWorkingGroup') {
+    const storageBucket = await overlay.getRepository(StorageBucket).getById(bucketId)
+    if (!storageBucket) {
+      return invalidMetadata(
+        SetNodeOperationalStatus,
+        `The storage bucket ${bucketId} does not exist`
+      )
+    } else if (storageBucket.operatorStatus.isTypeOf !== 'StorageBucketOperatorStatusActive') {
+      return invalidMetadata(
+        SetNodeOperationalStatus,
+        `The storage bucket ${bucketId} is not active`
+      )
+      // If the actor is a worker, check if the worker is the operator of the storage bucket
+    } else if (actor && storageBucket.operatorStatus.workerId !== actor.runtimeId) {
```

Comment:
If I understand correctly, this check will ensure that regular worker can only update status of their own bucket and not somebody's elses. Should we also add a check for when the lead is the actor that ensures the `meta.bucketId` and `meta.workerId` matches `storageBucket.operatorStatus.workerId`?

Reply:
Correct, there is no validation currently to check `meta.workerId = bucket.operatorStatus.workerId`. So the drawback of the lack of this validation is that in the case of lead remark the operation would be performed even though `meta.bucketId` and `meta.workerId` do not match for any bucket, so, mappings are just looking for any valid `meta.bucketId` and set the status on that, (Simply ignoring what is the value of `meta.workerId`). And this is fine for the Lead context I think.